syncthing-discovery/src/main/kotlin/net/syncthing/java/discovery/protocol/GlobalDiscoveryUtil.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2016 Davide Imbriaco
+ * Copyright (C) 2018 Jonas Lochmann
+ *
+ * This Java file is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.syncthing.java.discovery.protocol
+
+import com.google.gson.stream.JsonReader
+import kotlinx.coroutines.experimental.Dispatchers
+import kotlinx.coroutines.experimental.withContext
+import net.syncthing.java.core.beans.DeviceId
+import java.io.BufferedInputStream
+import java.io.IOException
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import java.security.SecureRandom
+import java.security.cert.X509Certificate
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.HttpsURLConnection
+import javax.net.ssl.SSLContext
+import javax.net.ssl.X509TrustManager
+
+object GlobalDiscoveryUtil {
+    suspend fun queryAnnounceServer(server: String, deviceId: DeviceId): AnnouncementMessage {
+        return withContext(Dispatchers.IO) {
+            val url = URL("https://$server/v2/?device=${deviceId.deviceId}")
```

Comment:
Maybe move the url pattern into a constant.

Reply:
This would prevent using the variables in it. I will move creating the url string to an extra function.